pkg/sql/explain_distsql.go

```diff
@@ -142,10 +158,17 @@ func (n *explainDistSQLNode) startExec(params runParams) error {
 		}
 	}
 
-	plan, err := distSQLPlanner.createPlanForNode(planCtx, n.plan)
+	var plan PhysicalPlan
+	var err error
+	if planNode, ok := n.plan.(distSQLExplainable); ok {
+		plan, err = planNode.makePlanForExplainDistSQL(planCtx, distSQLPlanner)
```

Comment:
This won't work in the case of something like `SELECT blah FROM [CREATE STATISTICS...]`, assuming `CREATE STATISTICS` returns a table, which is probably fine but it's worth pointing out, I think.